File: model/src/main/kotlin/com/testerum/model/text/parts/param_meta/StringTypeMeta.kt
package com.testerum.model.text.parts.param_meta

import com.fasterxml.jackson.annotation.JsonProperty

data class StringTypeMeta(@JsonProperty("javaType") override val javaType: String = "java.lang.String") : TypeMeta {

    override fun fileType(): String {
        return "TEXT"
    }

}

File: old/web/web-frontend/src/app/model/text/parts/param-meta/string-type.meta.ts
import {TypeMeta} from "./type-meta.model";
import {Serializable} from "../../../infrastructure/serializable.model";
import {JsonUtil} from "../../../../utils/json.util";

export class StringTypeMeta implements TypeMeta, Serializable<StringTypeMeta> {
    javaType: string;

    constructor(javaType = "java.lang.String") {
        this.javaType = javaType;
    }

    deserialize(input: Object): StringTypeMeta {
        this.javaType = input["javaType"];
        return this;
    }

    serialize(): string {
        return ''+
               '{' +
               '"@type": "TEXT"'+
               ',"javaType":'+ JsonUtil.stringify(this.javaType) +
               '}'
    }
}
